vector/src/main/java/im/vector/app/features/ftue/DefaultFTUEVariant.kt

```diff
@@ -0,0 +1,367 @@
+ * limitations under the License.
+ */
+
+package im.vector.app.features.ftue
+
+import android.content.Intent
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.ViewCompat
+import androidx.core.view.children
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentTransaction
+import com.airbnb.mvrx.withState
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import im.vector.app.R
+import im.vector.app.core.extensions.POP_BACK_STACK_EXCLUSIVE
+import im.vector.app.core.extensions.addFragment
+import im.vector.app.core.extensions.addFragmentToBackstack
+import im.vector.app.core.extensions.exhaustive
+import im.vector.app.core.platform.VectorBaseActivity
+import im.vector.app.databinding.ActivityLoginBinding
+import im.vector.app.features.home.HomeActivity
+import im.vector.app.features.login.LoginAction
+import im.vector.app.features.login.LoginCaptchaFragment
+import im.vector.app.features.login.LoginCaptchaFragmentArgument
+import im.vector.app.features.login.LoginConfig
+import im.vector.app.features.login.LoginFragment
+import im.vector.app.features.login.LoginGenericTextInputFormFragment
+import im.vector.app.features.login.LoginGenericTextInputFormFragmentArgument
+import im.vector.app.features.login.LoginMode
+import im.vector.app.features.login.LoginResetPasswordFragment
+import im.vector.app.features.login.LoginResetPasswordMailConfirmationFragment
+import im.vector.app.features.login.LoginResetPasswordSuccessFragment
+import im.vector.app.features.login.LoginServerSelectionFragment
+import im.vector.app.features.login.LoginServerUrlFormFragment
+import im.vector.app.features.login.LoginSignUpSignInSelectionFragment
+import im.vector.app.features.login.LoginSplashFragment
+import im.vector.app.features.login.LoginViewEvents
+import im.vector.app.features.login.LoginViewModel
+import im.vector.app.features.login.LoginViewState
+import im.vector.app.features.login.LoginWaitForEmailFragment
+import im.vector.app.features.login.LoginWaitForEmailFragmentArgument
+import im.vector.app.features.login.LoginWebFragment
+import im.vector.app.features.login.ServerType
+import im.vector.app.features.login.SignMode
+import im.vector.app.features.login.TextInputFormFragmentMode
+import im.vector.app.features.login.isSupported
+import im.vector.app.features.login.terms.LoginTermsFragment
+import im.vector.app.features.login.terms.LoginTermsFragmentArgument
+import im.vector.app.features.login.terms.toLocalizedLoginTerms
+import org.matrix.android.sdk.api.auth.registration.FlowResult
+import org.matrix.android.sdk.api.auth.registration.Stage
+import org.matrix.android.sdk.api.extensions.tryOrNull
+
+private const val FRAGMENT_REGISTRATION_STAGE_TAG = "FRAGMENT_REGISTRATION_STAGE_TAG"
+private const val FRAGMENT_LOGIN_TAG = "FRAGMENT_LOGIN_TAG"
+
+class DefaultFTUEVariant(
```

Comment:
this is a copy of https://github.com/vector-im/element-android/blob/develop/vector/src/main/java/im/vector/app/features/login/LoginActivity.kt